File: src/api/MoRS.ManagementSystem.Application/DTOs/Payment/PaymentResponse.cs
using MoRS.ManagementSystem.Application.DTOs.MembershipFee;
using MoRS.ManagementSystem.Application.DTOs.User;
using MoRS.ManagementSystem.Domain.Enums;
using System.Text.Json.Serialization;

namespace MoRS.ManagementSystem.Application.DTOs.Payment;

public class PaymentResponse
{
    public int Id { get; set; }

    public DateTime Date { get; set; }
    public decimal Amount { get; set; }
    public PaymentStatus Status { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public required string TransactionId { get; set; }
    public UserResponse User { get; set; } = null!;

    [JsonIgnore]
    public MembershipFeeResponse? MembershipFee { get; set; }
}

File: src/mobile/lib/models/payment/payment_response.dart
class PaymentResponse {
  final int id;
  final DateTime date;
  final double amount;
  final int status;
  final int paymentMethod;
  final String transactionId;
  final Map<String, dynamic>? user;

  PaymentResponse({
    required this.id,
    required this.date,
    required this.amount,
    required this.status,
    required this.paymentMethod,
    required this.transactionId,
    this.user,
  });

  factory PaymentResponse.fromJson(Map<String, dynamic> json) {
    return PaymentResponse(
      id: json['id'] as int,
      date: DateTime.parse(json['date'] as String),
      amount: (json['amount'] as num).toDouble(),
      status: json['status'] as int,
      paymentMethod: json['paymentMethod'] as int,
      transactionId: json['transactionId'] as String,
      user: json['user'] as Map<String, dynamic>?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'date': date.toIso8601String(),
      'amount': amount,
      'status': status,
      'paymentMethod': paymentMethod,
      'transactionId': transactionId,
      'user': user,
    };
  }
}
